extensions/iam.tf

```diff
@@ -0,0 +1,169 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+resource "aws_iam_role" "codebase_pipeline_deploy" {
+  name               = "${var.args.application}-${var.environment}-codebase-pipeline-deploy"
+  assume_role_policy = data.aws_iam_policy_document.assume_codebase_pipeline.json
+  tags               = local.tags
+}
+
+data "aws_iam_policy_document" "assume_codebase_pipeline" {
+  statement {
+    effect = "Allow"
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${var.args.pipeline_account_id}:root"]
+    }
+    condition {
+      test = "StringLike"
```

Comment:
Probably make little difference, but you could use ArnLike here to make it a bit tighter